src/app/screens/swap/useAlexSponsoredTransaction.ts

```diff
@@ -24,8 +34,35 @@ export const useAlexSponsoredTransaction = (userOverrideSponsorValue: boolean) =
   const sponsorTransaction = async (signed: StacksTransaction) =>
     alexSDK.broadcastSponsoredTx(signed.serialize().toString('hex'));
 
-  const { data: stxPendingTxData } = useStxPendingTxData();
-  const hasPendingTransactions = stxPendingTxData?.pendingTransactions?.length > 0;
+  const getAccountCurrentNonce = async () => {
+    let nonce = 0;
+    const confirmedTransactions: StxTransactionListData = await getConfirmedTransactions({
+      stxAddress,
+      network: selectedNetwork,
+    });
+
+    if (confirmedTransactions && confirmedTransactions.transactionsList.length > 0) {
+      nonce = confirmedTransactions.transactionsList[0].nonce + 1;
+    }
+    const currentPendingNonce =
+      (stxPendingTxData?.pendingTransactions ?? []).reduce(
+        (maxNonce, transaction) => Math.max(maxNonce, transaction?.nonce ?? 0),
+        0,
+      ) + 1;
+
+    return Math.max(currentPendingNonce, nonce);
+  };
+
+  useEffect(() => {
+    (async () => {
+      const currentNonce = await getAccountCurrentNonce();
+      const filteredStxPendingTxData = stxPendingTxData?.pendingTransactions.filter(
+        (transaction) => currentNonce - transaction?.nonce === 1,
```

Comment:
likewise, think about where this logic should belong.
is it likely to be reused in future? then put it in core
is it specific to ALEX's sponsor service? then leave it here, but it needs to be more clear why we filter like this. you can either use a concise comment, or better naming